pyproject.toml

```diff
@@ -1,3 +1,54 @@
+[project]
+name="spack"
+version="0"
```

Comment:
```suggestion
version="0.19.0.dev0"
```

Reply:
This is one part I wasn't sure about, it's the only place where this adds a *new* place to specify something that could get out of sync rather than removing one.  Probably still best to have it be right.

Reply:
Looks like setuptools supports dynamically getting the version, not sure about hatchling: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata